app/src/main/java/org/wikipedia/settings/LogoutPreference.kt

```diff
@@ -43,5 +45,18 @@ class LogoutPreference : Preference {
                     }.show()
             }
         }
+        holder.itemView.findViewById<Button>(R.id.accountVanishButton).setOnClickListener {
+            activity?.let {
+                MaterialAlertDialogBuilder(it, R.style.AlertDialogTheme_Icon_Delete)
+                    .setIcon(R.drawable.ic_person_remove)
+                    .setTitle(R.string.account_vanish_request_confirm_title)
+                    .setMessage(StringUtil.fromHtml(it.getString(R.string.account_vanish_request_confirm)))
+                    .setNegativeButton(android.R.string.cancel, null)
+                    .setPositiveButton(R.string.account_vanish_request_title) { _, _ ->
+                        it.finish()
```

Comment:
Not sure but is this intentional? Or should we just need to dismiss the dialog?

Reply:
This is intentional - when the user navigates "back" from the web form (whether they submitted it or not), we want them to get back to the Main screen, not the Settings screen, and this is the easiest way to do it.

Reply:
Got it. Thanks!